[2025-04-02T23:48:48.661+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Weather_Proj.Spark_Processing manual__2025-04-02T23:48:08.553036+00:00 [queued]>
[2025-04-02T23:48:48.669+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Weather_Proj.Spark_Processing manual__2025-04-02T23:48:08.553036+00:00 [queued]>
[2025-04-02T23:48:48.670+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-04-02T23:48:48.682+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): Spark_Processing> on 2025-04-02 23:48:08.553036+00:00
[2025-04-02T23:48:48.687+0000] {standard_task_runner.py:60} INFO - Started process 5807 to run task
[2025-04-02T23:48:48.691+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'Weather_Proj', 'Spark_Processing', 'manual__2025-04-02T23:48:08.553036+00:00', '--job-id', '60', '--raw', '--subdir', 'DAGS_FOLDER/Project_tasks.py', '--cfg-path', '/tmp/tmps2bizw0i']
[2025-04-02T23:48:48.692+0000] {standard_task_runner.py:88} INFO - Job 60: Subtask Spark_Processing
[2025-04-02T23:48:48.734+0000] {task_command.py:423} INFO - Running <TaskInstance: Weather_Proj.Spark_Processing manual__2025-04-02T23:48:08.553036+00:00 [running]> on host 520607ad0646
[2025-04-02T23:48:48.803+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='Weather_Proj' AIRFLOW_CTX_TASK_ID='Spark_Processing' AIRFLOW_CTX_EXECUTION_DATE='2025-04-02T23:48:08.553036+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-02T23:48:08.553036+00:00'
[2025-04-02T23:48:55.624+0000] {logging_mixin.py:188} INFO - root
 |-- data: struct (nullable = true)
 |    |-- base: string (nullable = true)
 |    |-- clouds: struct (nullable = true)
 |    |    |-- all: long (nullable = true)
 |    |-- cod: long (nullable = true)
 |    |-- coord: struct (nullable = true)
 |    |    |-- lat: double (nullable = true)
 |    |    |-- lon: double (nullable = true)
 |    |-- dt: long (nullable = true)
 |    |-- id: long (nullable = true)
 |    |-- main: struct (nullable = true)
 |    |    |-- feels_like: double (nullable = true)
 |    |    |-- grnd_level: long (nullable = true)
 |    |    |-- humidity: long (nullable = true)
 |    |    |-- pressure: long (nullable = true)
 |    |    |-- sea_level: long (nullable = true)
 |    |    |-- temp: double (nullable = true)
 |    |    |-- temp_max: double (nullable = true)
 |    |    |-- temp_min: double (nullable = true)
 |    |-- name: string (nullable = true)
 |    |-- sys: struct (nullable = true)
 |    |    |-- country: string (nullable = true)
 |    |    |-- id: long (nullable = true)
 |    |    |-- sunrise: long (nullable = true)
 |    |    |-- sunset: long (nullable = true)
 |    |    |-- type: long (nullable = true)
 |    |-- timezone: long (nullable = true)
 |    |-- visibility: long (nullable = true)
 |    |-- weather: array (nullable = true)
 |    |    |-- element: struct (containsNull = true)
 |    |    |    |-- description: string (nullable = true)
 |    |    |    |-- icon: string (nullable = true)
 |    |    |    |-- id: long (nullable = true)
 |    |    |    |-- main: string (nullable = true)
 |    |-- wind: struct (nullable = true)
 |    |    |-- deg: long (nullable = true)
 |    |    |-- gust: double (nullable = true)
 |    |    |-- speed: double (nullable = true)
[2025-04-02T23:48:55.723+0000] {logging_mixin.py:188} INFO - root
 |-- city_name: string (nullable = true)
 |-- city_timezone: long (nullable = true)
 |-- time_UTC: long (nullable = true)
 |-- city_country: string (nullable = true)
 |-- sunrise_time_UTC: long (nullable = true)
 |-- sunset_time_UTC: long (nullable = true)
 |-- weather_description: string (nullable = true)
 |-- sea_level_Meter: long (nullable = true)
 |-- temp_celisus: double (nullable = true)
 |-- temp_min_celisus: double (nullable = true)
 |-- temp_max_celisus: double (nullable = true)
 |-- humidity_gram/m3: long (nullable = true)
 |-- pressure_Pascal: long (nullable = true)
 |-- latitude: double (nullable = true)
 |-- longtiude: double (nullable = true)
[2025-04-02T23:48:55.745+0000] {java_gateway.py:2273} INFO - Callback Server Starting
[2025-04-02T23:48:55.746+0000] {java_gateway.py:2275} INFO - Socket listening on ('127.0.0.1', 46661)
[2025-04-02T23:48:58.550+0000] {clientserver.py:561} INFO - Python Server ready to receive messages
[2025-04-02T23:48:58.552+0000] {clientserver.py:575} INFO - Received command c on object id p0
[2025-04-02T23:48:58.556+0000] {clientserver.py:621} ERROR - There was an exception while executing the Python Proxy on the Python Side.
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/clientserver.py", line 617, in _call_proxy
    return_value = getattr(self.pool[obj_id], method)(*params)
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/utils.py", line 120, in call
    raise e
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/utils.py", line 117, in call
    self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
TypeError: write_to_cassandra() takes 1 positional argument but 2 were given
[2025-04-02T23:48:58.733+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/WeatherScripts/spark_Weather.py", line 123, in processData
    query.awaitTermination(60)
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/streaming/query.py", line 219, in awaitTermination
    return self._jsq.awaitTermination(int(timeout * 1000))
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/java_gateway.py", line 1322, in __call__
    return_value = get_return_value(
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/errors/exceptions/captured.py", line 185, in deco
    raise converted from None
pyspark.errors.exceptions.captured.StreamingQueryException: [STREAM_FAILED] Query [id = 28606eb8-127f-4e0b-b814-f05efc527dc0, runId = 63758ce8-de29-46d1-ac07-1eb7f27d2b98] terminated with exception: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/clientserver.py", line 617, in _call_proxy
    return_value = getattr(self.pool[obj_id], method)(*params)
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/utils.py", line 120, in call
    raise e
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/utils.py", line 117, in call
    self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
TypeError: write_to_cassandra() takes 1 positional argument but 2 were given

[2025-04-02T23:48:58.752+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=Weather_Proj, task_id=Spark_Processing, execution_date=20250402T234808, start_date=20250402T234848, end_date=20250402T234858
[2025-04-02T23:48:58.769+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 60 for task Spark_Processing ([STREAM_FAILED] Query [id = 28606eb8-127f-4e0b-b814-f05efc527dc0, runId = 63758ce8-de29-46d1-ac07-1eb7f27d2b98] terminated with exception: An exception was raised by the Python Proxy. Return Message: Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/py4j/clientserver.py", line 617, in _call_proxy
    return_value = getattr(self.pool[obj_id], method)(*params)
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/utils.py", line 120, in call
    raise e
  File "/home/airflow/.local/lib/python3.10/site-packages/pyspark/sql/utils.py", line 117, in call
    self.func(DataFrame(jdf, wrapped_session_jdf), batch_id)
TypeError: write_to_cassandra() takes 1 positional argument but 2 were given
; 5807)
[2025-04-02T23:48:58.794+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-02T23:48:58.814+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
